File: examples/console.cpp
#include <iostream>
#include <thread>

#include "crazyflieLinkCpp/Connection.h"

using namespace bitcraze::crazyflieLinkCpp;

int main()
{
    Connection con("radio://0/80/2M/E7E7E7E7E7");

    while (true) {
        Packet p = con.recv(0);
        if (p.port() == 0 && p.channel() == 0) {
            std::string str((const char*)p.payload(), (size_t)p.payloadSize());
            std::cout << str;
        }
    }

    return 0;
}

File: examples/console.py
#!/usr/bin/env python3
import cflinkcpp

if __name__ == "__main__":

  # connect to a CF
  con = cflinkcpp.Connection("radio://0/80/2M/E7E7E7E7E7")

  while True:
    p = con.recv(timeout=100)
    if p.valid and p.port == 0 and p.channel == 0:
      print(p.payload.decode('utf-8'), end='')
